File: C# Examples/08 Data Import and Export/3D Points/SolidWorksMacro.cs
using SolidWorks.Interop.sldworks;
using System.Runtime.InteropServices;

namespace _3D_Points
{
    public partial class SolidWorksMacro
    {
public void Main()
{
    ModelDoc2 swDoc = null;
    swDoc = ((ModelDoc2)(swApp.ActiveDoc));
    swDoc.SketchManager.Insert3DSketch(true);
    swDoc.SketchManager.AddToDB = true;

    ExcelDataReader edr = new ExcelDataReader();
    foreach (Point p in edr.getExcelPoints(SolidWorks.Interop.swconst.swLengthUnit_e.swINCHES))
    {
        swDoc.SketchManager.CreatePoint(p.X, p.Y, p.Z);
    }

    swDoc.SketchManager.InsertSketch(true);
    swDoc.SketchManager.AddToDB = false;
    swDoc.ClearSelection2(true);
    swDoc.ViewZoomtofit2();
 
}        

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

File: 08 Data Import and Export/3DPoints/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System


Partial Class SolidWorksMacro
    Public Sub main()
        Dim swDoc As ModelDoc2 = Nothing
        swDoc = CType(swApp.ActiveDoc, ModelDoc2)
        swDoc.SketchManager.Insert3DSketch(True)

        swDoc.SketchManager.AddToDB() = True

        Dim edr As New ExcelDataReader2
        Dim points As List(Of Point) = edr.getExcelPoints(swLengthUnit_e.swINCHES)
        For Each pt As Point In points
            swDoc.SketchManager.CreatePoint(pt.X, pt.Y, pt.Z)
        Next

        'For Each pt As Double() In getExcelPoints()
        '    swDoc.SketchManager.CreatePoint(pt(0), pt(1), pt(2))
        'Next

        swDoc.SketchManager.AddToDB() = False
        swDoc.ClearSelection2(True)
        swDoc.SketchManager.InsertSketch(True)
        swDoc.ViewZoomtofit2()
    End Sub

    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks
End Class
